src/System.Net.Security/src/System/Net/SecureProtocols/SslStreamInternal.cs

```diff
@@ -225,26 +316,34 @@ private void ProcessWrite(byte[] buffer, int offset, int count)
             }
             finally
             {
-                _NestedWrite = 0;
+                if (asyncRequest == null || failed)
```

Comment:
Thanks for pointing it out! It's good to see the pattern used in another tested code-path.